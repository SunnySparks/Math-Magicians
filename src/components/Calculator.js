import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min';
import '../App.css';
import React from 'react';
import calculate from '../logic/calculate';

// eslint-disable-next-line react/prefer-stateless-function
export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick = (e) => {
    this.setState((state) => calculate(state, e));
  }

  render() {
    return (
      <div className="container">
        <div className="row">
          <div className="col-lg-12 text-end bg-secondary text-white pb-5 pt-5">0</div>
        </div>
        <div className="row border-bottom border-secondary">
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="AC" onClick={() => { this.handleClick('AC'); }}>AC</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="+/-">+/-</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="%">%</button>
          <button className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5" type="button" name="รท">รท</button>
        </div>
        <div className="row border-bottom border-secondary">
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="7" onClick={() => { this.handleClick('7'); }}>7</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="8" onClick={() => { this.handleClick('8'); }}>8</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="9" onClick={() => { this.handleClick('9'); }}>9</button>
          <button className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5" type="button" name="x">x</button>
        </div>
        <div className="row border-bottom border-secondary">
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="4" onClick={() => { this.handleClick('4'); }}>4</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="5" onClick={() => { this.handleClick('5'); }}>5</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="6" onClick={() => { this.handleClick('6'); }}>6</button>
          <button className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5" type="button" name="-">-</button>
        </div>
        <div className="row border-bottom border-secondary">
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="1" onClick={() => { this.handleClick('1'); }}>1</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="2" onClick={() => { this.handleClick('2'); }}>2</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="3" onClick={() => { this.handleClick('3'); }}>3</button>
          <button className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5" type="button" name="+">+</button>
        </div>
        <div className="row">
          <button className="col-md-6 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="0" onClick={() => { this.handleClick('0'); }}>0</button>
          <button className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5" type="button" name="." onClick={() => { this.handleClick('.'); }}>.</button>
          <button className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5" type="button" name="=" onClick={() => { this.handleClick('='); }}>=</button>
        </div>
      </div>
    );
  }
}
