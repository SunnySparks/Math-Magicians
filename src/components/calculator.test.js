import Calculator from './Calculator';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import Link from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('displays the numbers and results when clicked', () => {
  render(<Calculator />)
  fireEvent.click(screen.getByText('7'));
  fireEvent.click(screen.getByText('+'));
  fireEvent.click(screen.getByText('7'));
  fireEvent.click(screen.getByText('='));
  screen.getByText('14');
});