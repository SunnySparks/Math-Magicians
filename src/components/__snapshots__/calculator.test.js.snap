// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="container pt-5"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 title-cont"
    >
      <h1>
        Let's do some math!
      </h1>
    </div>
    <div
      className="col-md-8 calculatrice"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12 text-end bg-white text-black pb-5 pt-5 border-secondary border-right"
        >
          0
          
          0
        </div>
      </div>
      <div
        className="row border-bottom border-secondary"
      >
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5"
          name="รท"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="row border-bottom border-secondary"
      >
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5"
          name="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="row border-bottom border-secondary"
      >
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5"
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="row border-bottom border-secondary"
      >
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5"
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="row"
      >
        <button
          className="col-md-6 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-grey pb-5 pt-5"
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="col-md-3 border-right border-secondary text-center bg-orange pb-5 pt-5"
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</nav>
`;
